blockchain/chainquery.go

```diff
@@ -77,7 +77,7 @@ type ChainTipInfo struct {
 // known chain tips in the block index.
 func (b *BlockChain) ChainTips() []ChainTipInfo {
 	b.index.RLock()
-	var chainTips []*blockNode
+	chainTips := make([]*blockNode, 0, len(b.index.chainTips))
```

Comment:
This isn't really right.  It's probably fine, but every entry in `b.index.chainTips` is itself of slice of `*blockNode` which means there are potentially multiple items per entry.  This is assuming there is only one entry per map entry which is almost never going to be a valid assumption.

Naturally, this will work without issues since the backing array will be grown as needed, however, I question whether it's worth using a valid that is certain to be incorrect.